Objects/moduleobject.c

```diff
@@ -836,15 +836,15 @@ _PyModuleSpec_IsUninitializedSubmodule(PyObject *spec, PyObject *name)
     return rc;
 }
 
-static int
-_get_file_origin_from_spec(PyObject *spec, PyObject **p_origin)
+int
+_PyModuleSpec_GetFileOrigin(PyObject *spec, PyObject **p_origin)
 {
     PyObject *has_location = NULL;
     int rc = PyObject_GetOptionalAttr(spec, &_Py_ID(has_location), &has_location);
     if (rc <= 0) {
         return rc;
     }
-    // If origin is not a location, or doesn't exist, or is not a str), we could consider falling
+    // If origin is not a location, or doesn't exist, or is not a str, we could consider falling
     // back to module.__file__. But the cases in which module.__file__ is not __spec__.origin
```

Comment:
@hauntsaninja I think this may be the source of the problem. On iOS, `module.__file__` isn't the same as `__spec__.origin` - module.__file__ is the original source of the binary module (in `lib-dynload`), in the sense of where Python thinks the library is installed; `__spec__.origin` is the actual iOS framework location (in the `Frameworks` folder).

Reply:
Ah interesting, I see the special importlib Loader. Thanks for fixing the test!